lib/SPIRV/libSPIRV/SPIRVFunction.cpp

```diff
@@ -136,6 +136,10 @@ bool SPIRVFunction::decodeBB(SPIRVDecoder &Decoder) {
       break;
     }
 
+    if (Decoder.OpCode == OpNop) {
+      continue;
```

Comment:
BTW, there is OpNop in SPIRV. Why we should not translate it?

Reply:
There is no `nop` instruction in LLVM IR, so there's nothing to translate it _to_, although there may be some benefit to decoding it.

With the decoding change above, `llvm-spirv -to-text` strips out `OpNop`, and `llvm-spirv -to-binary` still fails on `assert(Found && "Invalid key")` in `SPIRVMap::rmap()`.

The changes required for this in SPIRVReader are not obvious to me, but it seems to me that there is at least no need to support `OpNop` outside of function bodies, as in this regard, it is just an ordinary instruction.

Reply:
Sorry, I thought that we are translating from llvm ir to spirv here. Please disregard my comment.
I think a better solution might be if we add a case for OpNop here

https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/0905afb6c3388c629ff1050bbea31bb7afd1ad93/lib/SPIRV/SPIRVReader.cpp#L1506-L1511

Reply:
> I think a better solution might be if we add a case for OpNop here

There is currently no definition for `Nop` in `lib/SPIRV/libSPIRV/SPIRVInstruction.h`, so presumably that needs to be added. @AlexeySotkin, should this go under "Other instructions", i.e. should it be templated over `SPIRVInstTemplateBase` along with `SpecConstOp`, `GenericPtrMemSemantics` and `GenericCastToPtrExplicit`? These seem not to require any special capabilities / extensions and don't have any special validation.

Reply:
We have `OpNop` value here 
https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/2f667b52e89201cd4e1ae84ccf9e03da52946b39/lib/SPIRV/libSPIRV/spirv.hpp#L690
I think this should be enough for the purposes of this patch, right ?

Reply:
Currently the translator fails at the `Inst->getOpCode()` call in `SPIRVFunction::decodeBB()`, due to the `SPIRVDecoder::getEntry()` returning null on `OpCode == OpNop`. Once that is fixed, I expect `SPIRVEntry::create()` will need to create a `SPIRVNop` object.